javaslang/src/main/java/javaslang/collection/Traversable.java

```diff
@@ -447,6 +447,21 @@ default T single() {
     <C> Map<C, ? extends Traversable<T>> groupBy(Function<? super T, ? extends C> classifier);
 
     /**
+     * Matches each element with a unique key that you extract from it.
+     * If the same key is present twice, the function will return {@code None}.
+     *
+     * @param classifier A function which extracts a key from elements
+     * @param <K>        key class type
+     * @return A Map containing the elements arranged by their keys.
+     * @throws NullPointerException if {@code getKey} is null.
+     */
+    default <K> Option<Map<K, T>> arrangeBy(Function<? super T, K> getKey) {
+        return Option.of(groupBy(getKey).mapValues(Traversable<T>::singleOption))
```

Comment:
actually I wrote it with the <T> at first, but it doesn't build for me:

    [WARNING] /home/emmanuel/home/javaslang/javaslang/src/main/java/javaslang/collection/Traversable.java:[461,52] found raw type: javaslang.collection.Traversable
  missing type arguments for generic class javaslang.collection.Traversable<T>

and later:

    [ERROR] /home/emmanuel/home/javaslang/javaslang/src/main/java/javaslang/collection/Traversable.java: warnings found and -Werror specified

Maybe they removed the warning in a later version of java, but that sounds unlikely. Using 1.8.0_121.